shashi/practiceTest2.js

```diff
@@ -0,0 +1,67 @@
+//calculate Factorial of a number
+const calcFactorial = num => {
+  if(num < 0){
+    return `Not a value number to calculate factorial`;
+  } 
+  if (num === 0 || num === 1){
+    return 1;
+  } 
+  if(num > 1) {
+   return num * calcFactorial(num-1);
+  }
+}
+
+console.log(calcFactorial(5));
+
+//calculate power of a number
+const power = (num, times) => {
+  if(times < 0 || num < 0){
+    return -1;
+  }
+  if ( num === 0 || times === 0){
+    return 0;
+  } 
```

Comment:
thinking about how your code runs. 
- Do we need to check all the edges every time **power**? 
- What check do we really need for the recursion part? What don't we need?

- RY